src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -119,10 +119,13 @@ else if (!savedSenderID.equals(senderID)) {
                         editor.commit();
                     }
 
-                    if (gCachedExtras != null) {
+                    if (!gCachedExtras.isEmpty()) {
                         Log.v(LOG_TAG, "sending cached extras");
-                        sendExtras(gCachedExtras);
-                        gCachedExtras = null;
+                        Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
+                        while (gCachedExtrasIterator.hasNext()) {
+                            sendExtras(gCachedExtrasIterator.next());
```

Comment:
Instead of sending multiple notification events have you thought of creating a new method `sendExtras(ArrayList<Bundle> )` that would end up sending one `notification` event with an array of notification data objects?